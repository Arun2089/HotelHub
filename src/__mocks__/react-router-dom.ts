const mock = jest.requireActual("react-router-dom");

module.exports = {
  ...mock,
  useNavigate: jest.fn(),
  useParams: jest.fn(),
  useLocation: jest.fn(),
  useHistory: jest.fn(),
};
