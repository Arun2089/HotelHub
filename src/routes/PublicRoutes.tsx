import { useContext } from "react";
import { Navigate, Outlet } from "react-router-dom";
import { AuthContext } from "@context/AuthContext";

const PublicRoutes = () => {
  const authContext = useContext(AuthContext);
  if (authContext?.user) {
    return <Navigate to="/" />;
  }
  return <Outlet />;
};

export default PublicRoutes;
