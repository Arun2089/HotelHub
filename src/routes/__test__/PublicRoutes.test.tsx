import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter, useNavigate } from "react-router-dom";
import AuthProviderMock from "@src/test-utils/AuthContextMock";
import PublicRoutes from "../PublicRoutes";

const navigateMock = jest.fn();

describe("PublicRoutes", () => {
  it("snapshot test", async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthProviderMock value={{ isLoading: false }}>
          <PublicRoutes />
        </AuthProviderMock>
      </MemoryRouter>
    );
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it("renders Outlet when user is not authenticated", async () => {
    (useNavigate as jest.Mock).mockReturnValue(navigateMock);
    render(
      <MemoryRouter initialEntries={["/login"]}>
        <AuthProviderMock value={{ user: undefined }}>
          <PublicRoutes />
        </AuthProviderMock>
      </MemoryRouter>
    );
    expect(navigateMock).not.toHaveBeenCalled();
  });

  it("redirects to home when user is authenticated", async () => {
    (useNavigate as jest.Mock).mockReturnValue(navigateMock);
    render(
      <MemoryRouter initialEntries={["/login"]}>
        <AuthProviderMock value={{ isLoading: false }}>
          <PublicRoutes />
        </AuthProviderMock>
      </MemoryRouter>
    );
  });
});
