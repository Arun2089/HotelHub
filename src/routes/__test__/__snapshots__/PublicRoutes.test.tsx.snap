// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicRoutes snapshot test 1`] = `<DocumentFragment />`;
